azure-pipelines.yml

```
@@ -113,3 +113,52 @@ jobs:
     displayName: Run tests
 
   - script: conda clean --packages --yes
+
+- job: MPILinux
+  pool:
+    name: Default
+    demands:
+    - MultiGPU
+    - CUDA_VERSION
+    - Agent.OS -equals Linux
+  strategy:
+    matrix:
+      Python38:
+        python.version: '3.8'
+    maxParallel: 4
+  steps:
+
+  - script: echo "CUDA version is $(CUDA_VERSION)"
+    displayName: Print CUDA version
+
+  - script: conda update -n base conda --yes --quiet
+    displayName: Update conda
+
+  - script: rm /home/azure/conda/envs/tike -rf
+    displayName: Force remove previous environments
+
+  - script: >
+      conda create --quiet --yes
+      -n tike
+      --channel conda-forge
+      --file requirements.txt
+      pytest
+      python=$(python.version)
+      cudatoolkit=$(CUDA_VERSION)
+    displayName: Create build environment
+
+  - script: conda list -n tike
+    displayName: List build environment
+
+  - script: |
+      source activate tike
+      pip install . --no-deps
+    displayName: Setup and install
+
+  - script: |
+      source activate tike
+      mpirun -n 2 pytest tests/test_comm.py
+      mpirun -n 2 pytest tests/test_ptycho.py -k cgrad
+    displayName: Run tests
+
+  - script: conda clean --packages --yes
```

```suggestion
  - script: |
      source activate tike
      mpirun -n 2 pytest tests/test_comm.py
      mpirun -n 2 pytest tests/test_ptycho.py -k cgrad
    displayName: Run tests with MPI

  - script: conda clean --packages --yes
```
We don't need a new job specifically for MPILinux because it doesn't demand anything different from the existing MultiLinux job. We can just add a new step tot he MultiLinux job.

The reason we have a MultiLinux job and a Linux job is that MultiLinux demands >=2 GPUs (only one container meets this criteria), and the Linux only demands >=1 GPUs (multiple containers meet this criteria). Testing multi-gpu against only one Python version lets the tests complete faster.